pkg/backends/calico/routes.go

```diff
@@ -195,32 +196,49 @@ func (rg *routeGenerator) setRouteForSvc(svc *v1.Service, ep *v1.Endpoints) {
 	rg.Lock()
 	if rg.advertiseThisService(svc, ep) {
```

Comment:
Should there also be a `routes, exists := rg.svcRouteMap[key]; !exists || (routes != svc.Spec.ClusterIP + "/32")` condition here, to avoid some downstream churn?  (Or is that too complicated to be worth it?)

Reply:
Oh, and if `(routes != svc.Spec.ClusterIP + "/32")` is a possibility, I guess there should also be a `rg.client.DeleteStaticRoutes(routes)` in that case.

Reply:
(If that's not a possibility, it's perhaps worth simplifying the data structures and code to make that clearer.  I'm not sure I've understood what `key` is here, but it looks over-engineered for `rg.svcRouteMap` to be `map[key][]string` instead of just `map[key]string`.)

Reply:
yeah, the original intention was that we may also advertise externalIPs and load balancer IPs, but we haven't done that work yet.

I think we could simplify this for now like you suggest, and make it a list again if we ever need it, so I'm going to do that.

Reply:
I don't think it's possible for the stored route to differ from the calculated one, since `clusterIP` is an immutable field.

But, perhaps there are race conditions where we miss a delete for a service inbetween two adds? I'll add the defensive code anyway.